test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1675 @@
+    type = 1
+    flag = 17471
+Name: ssinodes
+    type = 1
+    flag = 1087
+Name: resc
+    type = 3
+    flag = 63
+Name: software
+    type = 3
+    flag = 63
+Name: site
+    type = 3
+    flag = 1087
+Name: exec_vnode
+    type = 3
+    flag = 61
+Name: start_time
+    type = 1
+    flag = 61
+Name: mpphost
+    type = 3
+    flag = 1087
+Name: mpparch
+    type = 3
+    flag = 1087
+Name: mpplabels
+    type = 3
+    flag = 1087
+Name: mppwidth
+    type = 1
+    flag = 1087
+Name: mppdepth
+    type = 1
+    flag = 1087
+Name: mppnppn
+    type = 1
+    flag = 1087
+Name: mppnodes
+    type = 3
+    flag = 3135
+Name: mppmem
+    type = 5
+    flag = 1087
+Name: mppt
+    type = 1
+    flag = 3135
+Name: partition
+    type = 3
+    flag = 1085
+Name: accelerator
+    type = 11
+    flag = 132159
+Name: accelerator_model
+    type = 3
+    flag = 132159
+Name: accelerator_memory
+    type = 5
+    flag = 181311
+Name: accelerator_group
+    type = 3
+    flag = 131135
+Name: |unknown|
+    type = 0
+    flag = 63
+"""
+
+# Define an enum which is used to label various pieces of information
+(   # qstat outputs
+    QSTAT_B_OUT,
+    QSTAT_BF_OUT,
+    QSTAT_OUT,
+    QSTAT_F_OUT,
+    QSTAT_T_OUT,
+    QSTAT_TF_OUT,
+    QSTAT_X_OUT,
+    QSTAT_XF_OUT,
+    QSTAT_NS_OUT,
+    QSTAT_FX_DSV_OUT,
+    QSTAT_F_DSV_OUT,
+    QSTAT_Q_OUT,
+    QSTAT_QF_OUT,
+    # qmgr outputs
+    QMGR_PS_OUT,
+    QMGR_PH_OUT,
+    QMGR_LPBSHOOK_OUT,
+    QMGR_LSCHED_OUT,
+    QMGR_PN_OUT,
+    QMGR_PR_OUT,
+    # pbsnodes outputs
+    PBSNODES_VA_OUT,
+    PBSNODES_A_OUT,
+    PBSNODES_AVSJ_OUT,
+    PBSNODES_ASJ_OUT,
+    PBSNODES_AVS_OUT,
+    PBSNODES_AS_OUT,
+    PBSNODES_AFDSV_OUT,
+    PBSNODES_AVFDSV_OUT,
+    # pbs_rstat outputs
+    PBS_RSTAT_OUT,
+    PBS_RSTAT_F_OUT,
+    # PBS config related outputs
+    PBS_CONF,
+    PBS_PROBE_OUT,
+    PBS_HOSTN_OUT,
+    PBS_ENVIRONMENT,
+    # System related outputs
+    OS_INFO,
+    PROCESS_INFO,
+    LSOF_PBS_OUT,
+    ETC_HOSTS,
+    ETC_NSSWITCH_CONF,
+    VMSTAT_OUT,
+    DF_H_OUT,
+    DMESG_OUT,
+    PS_LEAF_OUT,
+    # Logs
+    ACCT_LOGS,
+    SVR_LOGS,
+    SCHED_LOGS,
+    MOM_LOGS,
+    PG_LOGS,
+    COMM_LOGS,
+    # Daemon priv directories
+    SVR_PRIV,
+    MOM_PRIV,
+    SCHED_PRIV,
+    # Core file information
+    CORE_SCHED,
+    CORE_SERVER,
+    CORE_MOM,
+    # Miscellaneous
+    RSCS_ALL,
+    CTIME) = range(56)
+
+
+# Define paths to various files/directories with respect to the snapshot
+# server/
+QSTAT_B_PATH = os.path.join("server", "qstat_B.out")
+QSTAT_BF_PATH = os.path.join("server", "qstat_Bf.out")
+QMGR_PS_PATH = os.path.join("server", "qmgr_ps.out")
+QSTAT_Q_PATH = os.path.join("server", "qstat_Q.out")
+QSTAT_QF_PATH = os.path.join("server", "qstat_Qf.out")
+QMGR_PR_PATH = os.path.join("server", "qmgr_pr.out")
+RSCS_PATH = os.path.join("server", "rscs_all")
+# server_priv/
+SVR_PRIV_PATH = "server_priv"
+ACCT_LOGS_PATH = os.path.join("server_priv", "accounting")
```

Comment:
I'm afraid I didn't get what you are suggesting, could you please explain it a bit more?